bokehjs/test/unit/core/util/svg.ts

```diff
@@ -0,0 +1,108 @@
+
+function stringToHTML(str: string) {
+  const parser = new DOMParser()
+  const doc = parser.parseFromString(str, "text/html")
+  return doc.body
+}
+
+function compare_on_dom(fn: (ctx: CanvasRenderingContext2D) => void, svg: string, {width, height}: {width: number, height: number}) {
+  const canvas = document.createElement("canvas")
+  canvas.height = height
+  canvas.width = width
+
+  document.body.appendChild(canvas)
+
+  const ctx = canvas.getContext("2d") as CanvasRenderingContext2D
+
+  fn(ctx)
+  document.body.appendChild(stringToHTML(svg))
+}
+
+describe("SVGRenderingContext2d", () => {
+  it("should fill text correctly", () => {
+    const size = {width: 50, height: 50}
+    const ctx = new SVGRenderingContext2D(size)
+
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
+      _ctx.font = "normal 16px Times"
+      _ctx.fillText("TEST", 0, 16)
+    }
+    test(ctx)
+
+    const svg = ctx.get_serialized_svg()
+
+    compare_on_dom(test, svg, size)
+
+    expect(svg).to.be.equal(`<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="50" height="50"><defs/><text fill="#000000" stroke="none" font-family="Times" font-size="16px" font-style="normal" font-weight="normal" text-decoration="normal" x="0" y="16" text-anchor="start" dominant-baseline="alphabetic">TEST</text></svg>`)
+  })
+
+  it("should stroke text correctly", () => {
+    const size = {width: 50, height: 50}
+    const ctx = new SVGRenderingContext2D(size)
+
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
+      _ctx.font = "16px serif"
+      _ctx.strokeText("TEST", 0, 16)
+    }
+    test(ctx)
+
+    const svg = ctx.get_serialized_svg()
+    compare_on_dom(test, svg, size)
+
+    expect(svg).to.be.equal(`<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="50" height="50"><defs/><text fill="none" stroke="#000000" font-family="serif" font-size="16px" font-style="normal" font-weight="normal" text-decoration="normal" x="0" y="16" text-anchor="start" dominant-baseline="alphabetic" stroke-miterlimit="10" stroke-dasharray="">TEST</text></svg>`)
+  })
+
+  it("should draw arcs correctly", () => {
+    const size = {width: 150, height: 200}
+    const ctx = new SVGRenderingContext2D(size)
+
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
+      // Draw shapes
+      for (let i = 0; i <= 3; i++) {
+        for (let j = 0; j <= 2; j++) {
+          _ctx.beginPath()
+          const x             = 25 + j * 50                 // x coordinate
+          const y             = 25 + i * 50                 // y coordinate
+          const radius        = 20                          // Arc radius
+          const startAngle    = 0                           // Starting point on circle
+          const endAngle      = Math.PI + (Math.PI * j) / 2 // End point on circle
```

Comment:
I'm repeating myself, but this is not the naming convention that we use in bokehjs. This applies to the other PR as well.